sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpAnnotatedMessage.java

```diff
@@ -46,10 +49,27 @@ public AmqpAnnotatedMessage(AmqpMessageBody body) {
      * @param message used to create another instance of {@link AmqpAnnotatedMessage}.
      *
      * @throws NullPointerException if {@code message} or {@link AmqpAnnotatedMessage#getBody() body} is null.
+     * @throws UnsupportedOperationException if {@link AmqpMessageBodyType} is {@link AmqpMessageBodyType#SEQUENCE} or
+     * {@link AmqpMessageBodyType#VALUE}.
+     * @throws IllegalStateException for invalid {@link AmqpMessageBodyType}.
```

Comment:
This documentation is really unclear. You're telling the user what will happen, but you aren't tell them why or how to avoid the issue.

Reply:
Okay  added more doc and code sample to explain more.